src/goto-programs/builtin_functions.cpp

```diff
@@ -636,6 +636,81 @@ void goto_convertt::do_enum_is_in_range(
   copy(assignment, ASSIGN, dest);
 }
 
+void goto_convertt::do_havoc_object_slice(
+  const exprt &lhs,
+  const symbol_exprt &function,
+  const exprt::operandst &arguments,
+  goto_programt &dest)
+{
+  std::string identifier = CPROVER_PREFIX "havoc_object_slice";
+
+  auto source_loc = function.find_source_location();
+  source_loc.add_pragma("disable:pointer-check");
+  source_loc.add_pragma("disable:pointer-primitive-check");
+
+  // check # arguments
+  if(arguments.size() != 2)
+  {
+    error().source_location = source_loc;
+    error() << "'" << CPROVER_PREFIX "havoc_object_slice"
+            << "' expected to have two arguments" << eom;
+    throw 0;
+  }
+
+  // check argument types
+  if(arguments[0].type().id() != ID_pointer)
+  {
+    error().source_location = source_loc;
+    error() << "'" << CPROVER_PREFIX "havoc_object_slice"
+            << "' first argument expected to have `void *` type" << eom;
+    throw 0;
+  }
+
+  if(arguments[1].type().id() != ID_unsignedbv)
```

Comment:
To improve readability, could we rewrite this as:

```suggestion
  if(arguments[1].type().id() != size_type().id())
```

Reply:
The idea with this check was to allow users that use the primitive directly from GOTO to use any `unsignedbv` type and not just `size_t` to specify sizes. This makes it more general than its C interface.

Reply:
Hmm, thinking more, I think it could actually be:

```suggestion
  if(arguments[1].type() != size_type())
```

We define the signature of `__CPROVER_havoc_slice` as:

```c
 void __CPROVER_havoc_slice(void *p, __CPROVER_size_t size);
```
for the frontend.

So, shouldn't the check here match the this signature?

Yes it should be an `unsignedbv` type, but I think it shouldn't be _any_ `unsignedbv` type. It should match the `size_t` semantics on the machine, which is computed by `size_type()`:

https://github.com/diffblue/cbmc/blob/48893287099cb5780302fe9dc415eb6888354fd6/src/util/c_types.cpp#L58-L72

Or am I missing something here?